cwltool/builder.py

```diff
@@ -296,18 +301,19 @@ def bind_input(self,
                     if "secondaryFiles" not in datum:
                         datum["secondaryFiles"] = []
                     for sf in aslist(schema["secondaryFiles"]):
-                        if isinstance(sf, dict) and 'pattern' in sf:
+                        if isinstance(sf, MutableMapping) and 'pattern' in sf:
                             if 'required' in sf:
                                 sf_required = sf['required']
                             else:
                                 sf_required = True
                             sf = sf['pattern']
                         else:
                             sf_required = True
-                        if isinstance(sf, dict) or "$(" in sf or "${" in sf:
+                        if isinstance(sf, MutableMapping) or "$(" in sf or "${" in sf:
                             sfpath = self.do_eval(sf, context=datum)
                         else:
-                            if sf.endswith('?'):
+                            if sf.endswith('?') and \
+                               self.cwl_version != "v1.0":
```

Comment:
how about `self.cwl_version is in ['v1.1.0-dev1']`? Seems safer

Reply:
Yup, I was considering the same thing, I was just reluctant because with each new version "v1.1", "v1.2", "v2.0" the list will need to be updated.  Now that I think about it, that's why we have conformance tests, so it will probably be covered with tests, so I'll change this :)